detox/src/devices/common/drivers/android/tools/HashHelper.js

```diff
@@ -0,0 +1,64 @@
+const crypto = require('crypto');
+const fs = require('fs');
+
+const { DetoxInternalError } = require('../../../../../errors');
+const path = require('path');
+
+const HASH_PATH = '/data/local/tmp/detox';
+
+class HashHelper {
+  constructor(adb, hashXfer, hashPath = HASH_PATH) {
+    this._adb = adb;
+    this._hashXfer  = hashXfer;
+    this._hashPath = hashPath;
+  }
+
+  async generateHash(path) {
+    if (!path) {
+      throw new DetoxInternalError(`Path must be provided for hash generation`);
+    }
+
+    return new Promise((resolve, reject) => {
```

Comment:
As an `async` function - it might be better, semantically, to `await` the promise rather than returning it